src/plugins/unified_field_list/public/hooks/use_grouped_fields.ts

```diff
@@ -0,0 +1,270 @@
+        setDataView(await services.dataViews.get(dataViewId));
+      }
+    };
+    getDataView();
+    // if field existence information changed, reload the data view too
+  }, [dataViewId, services.dataViews, setDataView, hasFieldDataHandler]);
+
+  const unfilteredFieldGroups: FieldListGroups<T> = useMemo(() => {
+    const containsData = (field: T) => {
+      if (!dataViewId || !dataView) {
+        return true;
+      }
+      const overallField = dataView.getFieldByName?.(field.name);
+      return Boolean(overallField && hasFieldDataHandler(dataViewId, overallField.name));
+    };
+
+    const fields = allFields || [];
+    const allSupportedTypesFields = onSupportedFieldFilter
+      ? fields.filter(onSupportedFieldFilter)
+      : fields;
+    const sortedFields = [...allSupportedTypesFields].sort(sortFields);
+    const groupedFields = {
+      ...getDefaultFieldGroups(),
+      ...groupBy(sortedFields, (field) => {
+        if (field.type === 'document') {
+          return 'specialFields';
+        } else if (dataView?.metaFields?.includes(field.name)) {
+          return 'metaFields';
+        } else if (containsData(field)) {
+          return 'availableFields';
+        } else return 'emptyFields';
+      }),
+    };
+    const selectedFields = onSelectedFieldFilter ? sortedFields.filter(onSelectedFieldFilter) : [];
+
+    let fieldGroupDefinitions: FieldListGroups<T> = {
+      SpecialFields: {
+        fields: groupedFields.specialFields,
+        fieldCount: groupedFields.specialFields.length,
+        isAffectedByGlobalFilter: false,
+        isAffectedByTimeFilter: false,
+        isInitiallyOpen: false,
+        showInAccordion: false,
+        title: '',
+        hideDetails: true,
+      },
+      SelectedFields: {
+        fields: selectedFields,
+        fieldCount: selectedFields.length,
+        isInitiallyOpen: true,
+        showInAccordion: true,
+        title: i18n.translate('unifiedFieldList.useGroupedFields.selectedFieldsLabel', {
+          defaultMessage: 'Selected fields',
+        }),
+        isAffectedByGlobalFilter: false,
+        isAffectedByTimeFilter: true,
+        hideDetails: false,
+        hideIfEmpty: true,
+      },
+      AvailableFields: {
+        fields: groupedFields.availableFields,
+        fieldCount: groupedFields.availableFields.length,
+        isInitiallyOpen: true,
+        showInAccordion: true,
+        title:
+          dataViewId && fieldsExistenceInfoUnavailable
+            ? i18n.translate('unifiedFieldList.useGroupedFields.allFieldsLabel', {
+                defaultMessage: 'All fields',
+              })
+            : i18n.translate('unifiedFieldList.useGroupedFields.availableFieldsLabel', {
+                defaultMessage: 'Available fields',
+              }),
+        helpText: i18n.translate('unifiedFieldList.useGroupedFields.allFieldsLabelHelp', {
+          defaultMessage: 'Data view fields', // TODO: what text should be in here by default?
```

Comment:
@gchaps @ninoslavmiskovic What tooltip copy should we define for Discover page (or by default)? On Lens page we have:

Regular case:
<img width="472" alt="Screenshot 2022-10-27 at 11 44 29" src="https://user-images.githubusercontent.com/1415710/198251921-566e7536-648b-4d76-bcf6-7f9cfed7ffaa.png">

For text-based queries:
<img width="447" alt="Screenshot 2022-10-27 at 11 46 49" src="https://user-images.githubusercontent.com/1415710/198252180-a794f9c9-a826-418d-aa20-838aebef461a.png">

Reply:
@gchaps I am totally okay with you taking a lead on this one and alter the copy so it does not have features we do not have in Discover yet e.g drag and drop 😊

Reply:
IMHO, the first tooltip is too long.  I feel that the tooltip should focus on why the user would want to add the field to the document table.  Here is a suggestion:

> Add fields to the document table to view values that match the current query and time range.

Do we need a second tooltip for text-based queries?

Reply:
Are we calling it "data table" or "document table"?

Should the tooltip just explain what an available field is?

> Available fields contain values based on your query.

> Available fields contain values based on the results of your query.

If we also want to explain how to add an available field to the data table:

> Your query returned values for these fields.  Click + to add an available field to the data table.

Reply:
I like this:
`Your query returned values for these fields. Click + to add an available field to the data table.`
And I think "data table" is better then "document table" since it covers it all. 
Another option could be 
`Your query returned values for these fields. Click + to add an available field to the table.`

Reply:
Let's go with this:

> Your query returned values for these fields. Click + to add an available field to the data table.

where + is the actual icon (f it looks ok).

Reply:
Thanks @gchaps and @kertal ! 
I will add it in my next PR for Discover page https://github.com/elastic/kibana/pull/144045